src/main/java/org/zaproxy/zap/extension/jwt/fuzzer/JWTFuzzerHandler.java

```diff
@@ -0,0 +1,158 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2020 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.jwt.fuzzer;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.SortedSet;
+import java.util.TreeSet;
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.network.HttpMessage;
+import org.parosproxy.paros.view.View;
+import org.zaproxy.zap.extension.fuzz.FuzzerOptions;
+import org.zaproxy.zap.extension.fuzz.httpfuzzer.HttpFuzzer;
+import org.zaproxy.zap.extension.fuzz.httpfuzzer.HttpFuzzerHandler;
+import org.zaproxy.zap.extension.fuzz.httpfuzzer.HttpFuzzerHandlerOptionsPanel;
+import org.zaproxy.zap.extension.fuzz.httpfuzzer.HttpFuzzerMessageProcessor;
+import org.zaproxy.zap.extension.fuzz.httpfuzzer.HttpFuzzerMessageProcessorCollection;
+import org.zaproxy.zap.extension.fuzz.httpfuzzer.HttpFuzzerMessageProcessorUI;
+import org.zaproxy.zap.extension.fuzz.httpfuzzer.HttpFuzzerMessageProcessorUIHandler;
+import org.zaproxy.zap.extension.fuzz.httpfuzzer.HttpFuzzerOptions;
+import org.zaproxy.zap.extension.fuzz.impl.FuzzerDialog;
+import org.zaproxy.zap.extension.fuzz.messagelocations.MessageLocationReplacement;
+import org.zaproxy.zap.extension.fuzz.messagelocations.MessageLocationReplacementGenerator;
+import org.zaproxy.zap.extension.fuzz.messagelocations.MessageLocationReplacer;
+import org.zaproxy.zap.extension.fuzz.messagelocations.MessageLocationReplacers;
+import org.zaproxy.zap.extension.fuzz.messagelocations.MessageLocationsReplacementStrategy;
+import org.zaproxy.zap.extension.fuzz.messagelocations.MultipleMessageLocationsBreadthFirstReplacer;
+import org.zaproxy.zap.extension.fuzz.messagelocations.MultipleMessageLocationsDepthFirstReplacer;
+import org.zaproxy.zap.extension.fuzz.messagelocations.MultipleMessageLocationsReplacer;
+import org.zaproxy.zap.extension.fuzz.payloads.PayloadGeneratorMessageLocation;
+import org.zaproxy.zap.extension.jwt.fuzzer.messagelocations.JWTMessageLocation;
+import org.zaproxy.zap.view.messagecontainer.SelectableContentMessageContainer;
+
+/** @author preetkaran20@gmail.com */
+public class JWTFuzzerHandler extends HttpFuzzerHandler {
```

Comment:
The problems which i faced in extending HttpFuzzer are:
1. There is no way to pass messageLocationReplacer to the methods so instead of creating instance in methods take the instances from the getters so that child can extend those methods and logic need not to be copied.